### PR TITLE
[legacy] Use the Same Python Everywhere

### PR DESCRIPTION
**Pull request by ChristianTackeGSI**
Use CMake tooling to find a good Python version and use it around.

Geant4 with Python needs boost-python.
So tell boost explicitly about the chosen Python version.

Also remove python 2.x from Debian/Ubuntu setup scripts.

NOTE: Current containers have not yet been rebuild with this.

### CODE REVIEW

File: cmake/legacy.cmake [hunk 1]
```diff
@@ -55,11 +55,11 @@ if(ICU_ROOT)
   set(icu "-DICU_ROOT=${ICU_ROOT}")
   set(boost_icu_config "--with-icu=${ICU_ROOT}")
 endif()
-if(PYTHON_EXECUTABLE)
-  set(python "-DPYTHON_EXECUTABLE=${PYTHON_EXECUTABLE}")
-  get_filename_component(PYTHON_EXECUTABLE_FILE "${PYTHON_EXECUTABLE}" NAME)
-  set(boost_python_config "--with-python=${PYTHON_EXECUTABLE_FILE}")
+if(NOT PYTHON_EXECUTABLE)
+  find_package(Python 3.0 REQUIRED)
```
[the remark this answers is not in the document]
Comment (dennisklein) [reply cmake/legacy.cmake:59]: Or as a question, what is the argument for using `REQUIRED`?

[the remark this answers is not in the document]
Comment (ChristianTackeGSI) [reply cmake/legacy.cmake:59]: The argument goes as follows: If we don't give boost an explicit hint, which Python to use, it either uses python2.x or none (at least on Ubuntu:rolling). But geant4 wants boost with python3.

So the idea is to enforce a consistant Python Version usage around.

Alternatively, we could look into disabling python in geant4?

**Comment by ChristianTackeGSI** [comment]
Side note: I still have no idea, what's going wrong on macos and rockylinux.

File: cmake/legacy.cmake (hunk 1 above)
[the remark this answers is not in the document]
Comment (dennisklein) [reply cmake/legacy.cmake:59]: Optimistically configuring all packages to use the same python, I totally agree, but my concern is the hard dependency on CMake's python detection in the superbuild logic (Yes, you could say, that Geant4 will depend on that logic anyways, but we don't really want to know this at this point, right?).

**Comment by ChristianTackeGSI** [comment]
Okay, looks like Rockylinux is affected by broken PCH generation in GCC, and boost defaults to generate PCHs.

Let's try to disable PCH in boost.

I have no idea, if we want / need PCH, and what this might affect.

**Review by dennisklein — APPROVED** [review]

**Pull request merged**